docs/UpgradeGuide2.md

```diff
@@ -0,0 +1,529 @@
+# Upgrading ddtrace to 2.0
+
+## From 1.x to 2.0
+
+Upgrading `ddtrace` from 1.x to 2.0 introduces some breaking changes which are outlined below.
+
+[**Upgrade basic usage**](#2.0-upgrading-basic-usage)
+
+For users with a basic implementation (configuration file + out-of-the-box instrumentation), only minor changes to your configuration file are required: most applications take just minutes to update. Check out the following sections for a step-by-step guide.
+
+- [Requires Ruby 2.5+](#2.0-requires-ruby-2.5+)
+- [Extracts datadog-ci gem](#2.0-extracts-ci)
+- [Configuration changes](#2.0-configuration-changes)
+  - [Replace `use` with `instrument`](#2.0-use-instrument)
+  - [Type checking](#2.0-type-checking)
+  - [Propagation default](#2.0-propagation-default)
+  - [Options](#2.0-options)
+
+---
+
+[**Upgrade advanced usage**](#2.0-upgrading-advanced-usage)
+
+For users with an advanced implementation of `ddtrace` (custom instrumentation, sampling or processing behavior, etc), some additional changes may be required. See the following sections for details about what features changed and how to use them.
+
+- [Frozen String](#2.0-frozen-string)
+- [Tracing API](#2.0-tracing-api)
+- [Configuration API](#2.0-configuration-api)
+- [Log Correlation](#2.0-log-correlation)
+- [Distributed Tracing](#2.0-distributed-tracing)
+- [Transport](#2.0-transport)
+- [Sampling](#2.0-sampling)
+
+---
+
+[**Upgrade Instrumentations**](#2.0-upgrade-instrumentations)
+
+Outlines the changes for our instrumentations (ie: change of support policy and option behaviour).
```

Comment:
It's not merged yet, but it's safe to assume the support policy can be [linked here](https://github.com/DataDog/dd-trace-rb/blob/a4ee62e46947cb91cb53eb7beba22bf070d22101/docs/Compatibility.md#support-policy).

If you'd rather wait for that to be merged, or for me to update this upgrade guide if it gets merged first that's fine.

Reply:
>It's not merged yet, but it's safe to assume the support policy can be [linked here](https://github.com/DataDog/dd-trace-rb/blob/a4ee62e46947cb91cb53eb7beba22bf070d22101/docs/Compatibility.md#support-policy). (remove the branch blob id, of course)

I think mentioning support policy changes here is confusing, I remove it.